tests/e2e/operator/operator_test.go

```diff
@@ -17,67 +17,62 @@
 package operator
 
 import (
+	"context"
 	"fmt"
 	"path/filepath"
 	"regexp"
 	"strings"
 	"time"
 
+	"github.com/istio-ecosystem/sail-operator/api/v1alpha1"
 	. "github.com/istio-ecosystem/sail-operator/pkg/test/util/ginkgo"
 	"github.com/istio-ecosystem/sail-operator/pkg/test/util/supportedversion"
+	. "github.com/istio-ecosystem/sail-operator/tests/e2e/operator/util/gomega"
 	"github.com/istio-ecosystem/sail-operator/tests/e2e/operator/util/helm"
 	"github.com/istio-ecosystem/sail-operator/tests/e2e/operator/util/kubectl"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"istio.io/istio/pkg/ptr"
 )
 
-var istiodVersionRegex = regexp.MustCompile(`Version:"(\d+\.\d+(\.\d+|-\w+))`)
+var (
+	istiodVersionRegex = regexp.MustCompile(`Version:"(\d+\.\d+(\.\d+|-\w+))`)
+	deployment         = &appsv1.Deployment{}
+	crd                = &apiextensionsv1.CustomResourceDefinition{}
+	crdList            = &apiextensionsv1.CustomResourceDefinitionList{}
+	daemonset          = &appsv1.DaemonSet{}
+	cni                = &v1alpha1.IstioCNI{}
+	istio              = &v1alpha1.Istio{}
```

Comment:
I'm not sure about defining these variables here instead of each time we need them. One problem I see is that if any tests ever run in parallel, they will fail because of these global variables.

They also make some assertions harder to understand. For example, in the following code, it's not completely clear what `crd` is:
```
				Eventually(getObject).WithArguments(ctx, cl, key(crdName), crd).
					Should(HaveCondition(apiextensionsv1.Established, metav1.ConditionTrue), "Error getting Istio CRD")
```

If you specify the object directly instead of using a variable, it makes it a bit more cleare:
```
				Eventually(getObject).WithArguments(ctx, cl, key(crdName), &apiextensionsv1.CustomResourceDefinition{}).
					Should(HaveCondition(apiextensionsv1.Established, metav1.ConditionTrue), "Error getting Istio CRD")
```